cache/s3proxy/s3proxy.go

```diff
@@ -68,13 +69,23 @@ func New(
 		log.Fatalf("Failed to determine s3proxy credentials")
 	}
 
+	secure := !DisableSSL
+	tr, err := minio.DefaultTransport(secure)
+	if err != nil {
+		log.Fatalf("Failed to create default transport: %v", err)
+	}
+
+	tr.MaxIdleConns = MaxIdleConns
```

Comment:
The minio default for `MaxIdleConnsPerHost` is 16, which was causing excessive tls negotiation even with a modest concurrency level (~200 concurrent connections) 

https://github.com/minio/minio-go/blob/1e5fd8a9ce3588100e5be814f564fd78436aa5f4/transport.go#L51-L52

Reply:
Which values did you find the best?

Reply:
In our setup its rare for us to exceed 200 concurrent, as we tend to be very bursty, 0 -> 200 -> 0 within a very small window. I expect this to change as our baseline load increases though. 

We're using Pants and there are two settings that define the max concurrency per client remote_cache_rpc_concurrency / remote_store_rpc_concurrency at 128 each. We'll occasionally have more than one client making requests concurrently, but because they're generally very fast they often don't overlap given our current workload.